### DISCUSSION TITLE
Support for JSON on SQLite and MySQL?

### DISCUSSION THREAD

**Discussion by 3ynm** [jeremyevans/sequel#1843] [General]
Is there any way to use the JSON type on SQLite or MySQL? If not. are there plans for supporting it?

**Comment by AlexWayfer** [comment]
It can be added via external plugins.

For example, easily founded: https://rubygems.org/gems/sequel-mysql_json

**Comment by jeremyevans** [comment]
@AlexWayfer pointed out an external gem for the MySQL JSON support.  Note that SQLite does not have a JSON type, it only has JSON functions and operators.

In terms of support shipping with Sequel, I'm not against it.  Nobody has requested it so far.  MySQL is complicated by the MySQL/MariaDB split.  I haven't investigated the differences between the JSON support in MySQL and MariaDB.  For the SQLite JSON support, I think the functions and operators are similar to PostgreSQL, so it probably would not be difficult to port the pg_json_ops extension to SQLite.  Maybe it's possible to use the pg_json_ops extension with SQLite already, I haven't tried.
